docs/tutorials/gluon/info_gan.md

```diff
@@ -0,0 +1,436 @@
+            self.G.add(nn.Conv2DTranspose(64 * 8, 4, 1, 0, use_bias=False))
+            self.G.add(nn.BatchNorm())
+            self.G.add(nn.Activation('relu'))
+            self.G.add(nn.Conv2DTranspose(64 * 4, 4, 2, 1, use_bias=False))
+            self.G.add(nn.BatchNorm())
+            self.G.add(nn.Activation('relu'))
+            self.G.add(nn.Conv2DTranspose(64 * 2, 4, 2, 1, use_bias=False))
+            self.G.add(nn.BatchNorm())
+            self.G.add(nn.Activation('relu'))
+            self.G.add(nn.Conv2DTranspose(64, 4, 2, 1, use_bias=False))
+            self.G.add(nn.BatchNorm())
+            self.G.add(nn.Activation('relu'))
+            self.G.add(nn.Conv2DTranspose(3, 4, 2, 1, use_bias=False))
+            self.G.add(nn.Activation('tanh'))
+
+    def hybrid_forward(self, F, x):
+        x = self.prev(x)
+        x = F.reshape(x, (0, -1, 1, 1))
+        return self.G(x)
+```
+
+## Discriminator
+Define the Discriminator and Q model. The Q model shares many layers with the Discriminator. Its task is to estimate the code `c` for a given fake image.  It is used to maximize the lower bound to the mutual information.
+
+
+```python
+class Discriminator(gluon.HybridBlock):
+    def __init__(self, **kwargs):
+        super(Discriminator, self).__init__(**kwargs)
+        with self.name_scope():
+            self.D = nn.HybridSequential()
+            self.D.add(nn.Conv2D(64, 4, 2, 1, use_bias=False))
+            self.D.add(nn.LeakyReLU(0.2))
+            self.D.add(nn.Conv2D(64 * 2, 4, 2, 1, use_bias=False))
+            self.D.add(nn.BatchNorm())
+            self.D.add(nn.LeakyReLU(0.2))
+            self.D.add(nn.Conv2D(64 * 4, 4, 2, 1, use_bias=False))
+            self.D.add(nn.BatchNorm())
+            self.D.add(nn.LeakyReLU(0.2))
+            self.D.add(nn.Conv2D(64 * 8, 4, 2, 1, use_bias=False))
+            self.D.add(nn.BatchNorm())
+            self.D.add(nn.LeakyReLU(0.2))
+
+            self.D.add(nn.Dense(1024, use_bias=False), nn.BatchNorm(), nn.Activation(activation='relu'))
+       
+            self.prob = nn.Dense(1)
+            self.feat = nn.HybridSequential()
+            self.feat.add(nn.Dense(128, use_bias=False), nn.BatchNorm(), nn.Activation(activation='relu'))
+            self.category_prob = nn.Dense(n_categories)
+            self.continuous_mean = nn.Dense(n_continuous)
+            self.Q = nn.HybridSequential()
+            self.Q.add(self.feat, self.category_prob, self.continuous_mean)
+
+    def hybrid_forward(self, F, x):
+        x               = self.D(x)
+        prob            = self.prob(x)
+        feat            = self.feat(x)
+        category_prob   = self.category_prob(feat)
+        continuous_mean = self.continuous_mean(feat)
+        
+        return prob, category_prob, continuous_mean
+```
+
+The InfoGAN has the following layout.
+<img src="https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/info_gan/InfoGAN.png" style="width:800px;height:250px;">
```

Comment:
I think the architecture diagram should be mirror inverted - 

Latent&Noise . -> . Generator . ->  Fake Data( + Real Data) . -> . Discriminator . -> (Real/Fake probability) Q Neural Network . -> Estimation of c

The flow of the diagram should be from left to right

Reply:
I updated the diagram: https://github.com/NRauschmayr/web-data/blob/tutorial_infogan/mxnet/doc/tutorials/info_gan/InfoGAN.png Let me know if you think it should be further improved.

Reply:
Would be better if it showed that the output of the Discriminator is the probability distribution of real and fake, as opposed to it currently showing just "Real|Fake". But its a minor point, you can take a call on what would be better.